ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ViolationsByPolicyCategoryChart.tsx

```diff
@@ -0,0 +1,261 @@
+ * have been filtered by the user in the widget's legend.
+ */
+function zeroOutFilteredSeverities(
+    groups: AlertGroup[],
+    hiddenSeverities: Set<PolicySeverity>
+): AlertGroup[] {
+    return groups.map(({ group, counts }) => ({
+        group,
+        counts: counts.map(({ severity, count }) => ({
+            severity,
+            count: hiddenSeverities.has(severity) ? '0' : count,
+        })),
+    }));
+}
+
+function pluckSeverityCount(severity: PolicySeverity): (group: AlertGroup) => number {
+    return ({ counts }) => {
+        const severityCount = counts.find((ct) => ct.severity === severity)?.count ?? '0';
+        return -parseInt(severityCount, 10);
+    };
+}
+
+function sortByVolume(groups: AlertGroup[]) {
+    const sum = (a: number, b: number) => a + b;
+    return sortBy(groups, ({ counts }) => {
+        return -counts.map(({ count }) => parseInt(count, 10)).reduce(sum);
+    });
+}
+
+function sortBySeverity(groups: AlertGroup[]) {
+    return sortBy(groups, [
+        pluckSeverityCount('CRITICAL_SEVERITY'),
+        pluckSeverityCount('HIGH_SEVERITY'),
+        pluckSeverityCount('MEDIUM_SEVERITY'),
+        pluckSeverityCount('LOW_SEVERITY'),
+    ]);
+}
+
+export type LifecycleOption = 'ALL' | Exclude<LifecycleStage, 'BUILD'>;
+
+type CountsBySeverity = Record<PolicySeverity, Record<string, number>>;
+
+function getCountsBySeverity(groups: AlertGroup[]): CountsBySeverity {
+    const result = {
+        LOW_SEVERITY: {},
+        MEDIUM_SEVERITY: {},
+        HIGH_SEVERITY: {},
+        CRITICAL_SEVERITY: {},
+    };
+
+    groups.forEach(({ group, counts }) => {
+        result.LOW_SEVERITY[group] = 0;
+        result.MEDIUM_SEVERITY[group] = 0;
+        result.HIGH_SEVERITY[group] = 0;
+        result.CRITICAL_SEVERITY[group] = 0;
+
+        counts.forEach(({ severity, count }) => {
+            result[severity][group] = parseInt(count, 10);
+        });
+    });
+
+    return result;
+}
+
+function linkForViolationsCategory(
+    category: string,
+    searchFilter: SearchFilter,
+    lifecycle: LifecycleOption,
+    hiddenSeverities: Set<PolicySeverity>
+) {
+    const search: SearchFilter = {
+        ...searchFilter,
+        Category: category,
+    };
+
+    if (lifecycle !== 'ALL') {
+        search['Lifecycle Stage'] = lifecycle;
+    }
+    if (hiddenSeverities.size > 0) {
+        search.Severity = severitiesLowToCritical.filter((s) => !hiddenSeverities.has(s));
+    }
+
+    const queryString = getQueryString({
+        s: search,
+        sortOption: { field: 'Severity', direction: 'desc' },
+    });
+    return `${violationsBasePath}${queryString}`;
+}
+
+type ViolationsByPolicyCategoryChartProps = {
+    alertGroups: AlertGroup[];
+    sortType: SortTypeOption;
+    lifecycle: LifecycleOption;
+    searchFilter: SearchFilter;
+    hiddenSeverities: Set<PolicySeverity>;
+    setHiddenSeverities: (severities: Set<PolicySeverity>) => Promise<Config>;
+};
```

Comment:
Can you move the Props type to immediately precede the component function?

On the other side of the coin, it moves the following helper function and module scope variable to be together with the others.